src/tail.rs

```diff
@@ -0,0 +1,88 @@
+use anyhow::Result;
+use std::cmp::min;
+use std::fs::File;
+use std::io::{Read, Seek, SeekFrom};
+
+pub(crate) fn tail(file_path: &str, n: usize) -> Result<Vec<String>> {
+    let mut file = File::open(file_path)?;
+    let file_len = file.seek(SeekFrom::End(0))?;
+
+    if file_len == 0 {
+        return Ok(Vec::new());
+    }
+
+    const CHUNK_SIZE: usize = 8192;
+    let mut buffer = vec![0u8; CHUNK_SIZE];
+    let mut position = file_len;
+    let mut lines_found = 0;
+    let mut result_bytes = Vec::new();
+    let mut found_final_newline = false;
+
+    // Read backwards in chunks
+    while position > 0 && lines_found < n {
+        let chunk_size = min(CHUNK_SIZE, position as usize);
+        position -= chunk_size as u64;
+
+        file.seek(SeekFrom::Start(position))?;
+        file.read_exact(&mut buffer[..chunk_size])?;
+
+        for &byte in buffer[..chunk_size].iter().rev() {
+            result_bytes.push(byte);
+
+            if byte == b'\n' {
```

Comment:
Do you want to handle the \r\n newline case here too? It's rare to find on Linux but it shows up from time to time